### PR TITLE
Add some test infrastructure

### PR DESCRIPTION
**Pull request by JayH5**
See the commits for detailed information but this now does:
- Puppet (.pp file) lint checks
- Puppet syntax checks
- metadata.json lint checks
- Makes sure classes compile
- Cleans up a bunch of stuff
- Adds a "test" task that runs all syntax and spec tests
- Adds basic coverage tests (reporting 0% coverage)

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -60,6 +60,10 @@
 #
 # [*manage_service*]
 #   Whether or not to manage the state of the Marathon service with Puppet.
+#
+# [*service*]
+#   Service to notify when there is a config change. If manage_service is true
+#   then this should not be set.
```
Comment (jerith) [inline manifests/init.pp:66]: How does ordering and notification happen if we're managing the service?

Comment (JayH5) [inline manifests/init.pp:66]: We notify `Service['marathon']` when managing the service. A `notify =>` in puppet also `require`s the resource. So the config will only be added after the service is defined and the service will be notified when those config files change. Does that answer your question?

Comment (jerith) [inline manifests/init.pp:66]: Cool, thanks.

**Comment by jerith** [comment]
:+1:

**Pull request merged**